subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/RelocateMainSymbol.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.swift.tasks;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.specs.Spec;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputFiles;
+import org.gradle.api.tasks.SourceTask;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.internal.os.OperatingSystem;
+import org.gradle.process.ExecSpec;
+import org.gradle.util.CollectionUtils;
+
+import java.io.File;
+import java.util.Set;
+
+/**
+ * Relocates the symbol <code>_main</code> to <code>relocated_main</code> in an object file, so the
+ *
+ * @since 4.4
+ */
+@Incubating
+public class RelocateMainSymbol extends SourceTask {
+    private File mainObjectFile;
```

Comment:
Should this be a provider?

Reply:
I'm using it as a cache for calls to findMainObject()